### PR TITLE
Rewrite Activity Report into new ReactUI

### PR DESCRIPTION
**Pull request by sherrif10**
PR  https://github.com/I-TECH-UW/OpenELIS-Global-2/issues/758

CC @mozzy11   @mherman22  and others let me know what you think ?

In this PR , i refactored 3 reports into one activity Report.

https://github.com/I-TECH-UW/OpenELIS-Global-2/assets/48015736/f73a2246-8296-4566-8e02-0d45aa077c92

There are things i need to refactor here, especially on the button part, i was thinking of having one button with embedded three functions.

### CODE REVIEW

**Comment by sherrif10** [comment]
Hello @mozzy11 . Its ready to be reviewed.

**Comment by manishjha-04** [comment]
Hey @sherrif10 , I think your idea is great.  Just a quick suggestion from my perspective as a user: perhaps keeping all the reports separate was  a much better approach in terms of user interface. Just a humble thought!

File: frontend/src/components/reports/routine/ActivityReportByTest.js [hunk 1]
```diff
@@ -0,0 +1,183 @@
...
+    }
+  };
+  const getBaseParams = (reportType) => {
+    const { startDate, endDate } = reportFormValues;
+    const formattedStartDate = startDate ? formatDate(startDate) : '';
+    const formattedEndDate = endDate ? formatDate(endDate) : '';
+    let baseParams = '';
+    switch (reportType) {
+      case "test":
+        baseParams = `Routinereport=activityReportByTest&type=indicator&lowerDateRange=${formattedStartDate}&upperDateRange=${formattedEndDate}`;
```
Comment (manishjha-04) [inline frontend/src/components/reports/routine/ActivityReportByTest.js:64]:
```suggestion
        baseParams = `RoutineReport?report=activityReportByTest&type=indicator&lowerDateRange=${formattedStartDate}&upperDateRange=${formattedEndDate}`;
```

Comment (sherrif10) [reply frontend/src/components/reports/routine/ActivityReportByTest.js:64]: Thanks @manishjha-04   ohh i definately understand. this was @mozzy11  idea. i can definately revert it  back.

File: frontend/src/components/reports/Routine.js [hunk 2]
```diff
@@ -56,23 +56,9 @@ export const RoutineReportsMenu = {
...
-            config.serverBaseUrl +
-            "/Report?type=indicator&report=activityReportByTest",
+          link:"/RoutineReport?type=indicator&report=activityReportByTest",
           label: <FormattedMessage id="sideNav.label.bytesttype" />,
         },
-        {
-          link:
-            config.serverBaseUrl +
-            "/Report?type=indicator&report=activityReportByPanel",
-          label: <FormattedMessage id="sideNav.label.bypaneltype" />,
```
Comment (mozzy11) [inline frontend/src/components/reports/Routine.js:68]: @sherrif10 you dont have to remove the menu items. you only had to refactor the report componenet

Comment (sherrif10) [reply frontend/src/components/reports/Routine.js:68]: https://github.com/I-TECH-UW/OpenELIS-Global-2/assets/48015736/f2df03b9-a7a6-41cf-900d-acd6af1f2448

Restored it back onto three components.

File: frontend/src/components/reports/routine/ActivityReportByPanel.js [hunk 3]
```diff
@@ -0,0 +1,148 @@
...
+                  </h5>
+                </Section>
+                <div className="inlineDiv">
+                  <CustomDatePicker
+                    id={"startDate"}
+                    labelText={intl.formatMessage({
+                      id: "eorder.date.start",
+                      defaultMessage: "Start Date",
+                    })}
+                    autofillDate={true}
```
Comment (mozzy11) [inline frontend/src/components/reports/routine/ActivityReportByPanel.js:106]: @sherrif10 youll realise all the three reports have alot of simillar componenets ie date inputs. 

Just have a single component and render the reports specific selector based on the report type

**Review by mozzy11 — CHANGES_REQUESTED** [review]
see above components

File: frontend/src/components/reports/routine/ActivityReportByPanel.js (hunk 3 above)
Comment (sherrif10) [reply frontend/src/components/reports/routine/ActivityReportByPanel.js:106]: Alright Thanks.What is your suggestion about this output. Is this how its supposed to be.

https://github.com/I-TECH-UW/OpenELIS-Global-2/assets/48015736/f73a2246-8296-4566-8e02-0d45aa077c92

File: frontend/src/components/reports/routine/Index.js [hunk 4]
```diff
@@ -78,6 +80,12 @@ const RoutineIndex = () => {
             {type === "indicator" && report === "statisticsReport" && (
               <StatisticsReport />
             )}
+            {type === "indicator" && report === "activityReportByTest" && (
```
[the remark this answers is not in the document]
Comment (sherrif10) [reply frontend/src/components/reports/routine/Index.js:84]: I updated this pr with each component rendered basing on report type.

File: frontend/src/components/reports/routine/ActivityReportByPanel.js (hunk 3 above)
Comment (sherrif10) [reply frontend/src/components/reports/routine/ActivityReportByPanel.js:106]: Basing on your response, i will have only one component. ActivityReport.

**Comment by sherrif10** [comment]
@mozzy11  let me know your suggestions about this current status of the pr.

**Comment by mozzy11** [comment]
@sherrif10 can you see the build failure ??

**Comment by mozzy11** [comment]
Thanks @sherrif10 , 
Can you borrow a leaf from this PR  https://github.com/I-TECH-UW/OpenELIS-Global-2/pull/912/files?
There two related reports and all are writen with in a sinfle component

**Comment by sherrif10** [comment]
Okay cool. thanks @mozzy11  i will update this pr as soon as possible.

**Pull request closed without merging**